### PR TITLE
Partial fix for PDF doc build

### PR DESCRIPTION
**Pull request by pushfoo**
TL;DR: replaced 403ing images + added gif converter, but LaTex still broken

### Changes

* Replaced missing (403) links starting with `camo.githubusercontent`
  * When screenshots were available, I used those
  * For when they weren't, there's a pngcrushed placeholder image
* Added `imageconvert` extension for PDF to convert gifs
* Cleaned up Boxhead Survivor sampel game links + made them clickable
* Added LaTeX build requirements notes

#### PDF Build: Still Broken

Running `./make.py latexpdf` produces plenty of warnings and ends with this error and a LaTeX input prompt:
```
LaTeX Warning: Hyper reference `tutorials/platform_tutorial/index:platformer-tu
torial' on page 67 undefined on input line 2403.

[67 <./YI7uLX.jpg>] [68 <./screenshot1.png>]
Runaway argument?
{{example
! Paragraph ended before \\sphinxsafeincludegraphics was complete.
<to be read again> 
                   \par 
l.2483 
       
? 
```

#### EPUB build: Untested

Haven't gotten to this yet.

### CODE REVIEW

**Comment by pushfoo** [comment]
The test failures appear to be `pyright` upset about the GUI.

File: doc/example_code/game_jam_2020.rst [hunk 1]
```diff
@@ -30,7 +30,7 @@ Here are the games that made it to the top 10:
 
     2nd Place: `Triple Blocks <https://github.com/python-discord/game-jam-2020/tree/master/Finalists/gamer_gang>`_
 
-.. figure:: https://camo.githubusercontent.com/f28f09b33244a7db72df7b4985791d89358f8d88/68747470733a2f2f63646e2e646973636f72646170702e636f6d2f6174746163686d656e74732f3639333137373530373638333336393031302f3730343030363031343535343437323439382f756e6b6e6f776e2e706e67
+.. figure:: /images/image_not_available.png
```
Comment (pvcraven) [inline doc/example_code/game_jam_2020.rst]: I'd rather just remove the image than sub in a 'not found' placeholder.

Comment (pushfoo) [reply doc/example_code/game_jam_2020.rst]: Removed + removed `image_not_avialable.png`

**Comment by pushfoo** [comment]
Rebased onto development to include #2062.
